cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java

```diff
@@ -1207,6 +1211,88 @@ private void testMultipleJoiner(Engine engine) throws Exception {
     validateMetric(3, appId, sinkName + ".records.in");
   }
 
+  /**
+   * Tests the secure storage macro function in MapReduce pipelines by creating datasets from the secure store data.
+   */
+  @Test
+  public void testSecureStoreMacroMapReduce() throws Exception {
```

Comment:
this test is the same as the spark one except for a setEngine() call. Can refactor to have a testSecureMacro() test that takes the engine as an argument, then the two tests just call that.